clients/src/main/java/org/apache/kafka/clients/consumer/events/AbstractKafkaServerEventData.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.consumer.events;
+
+abstract public class AbstractKafkaServerEventData {
+    private long timestampMs;
+    private KafkaServerEventType type;
```

Comment:
Should these be `public final` for direct access? As it is now, I don't think this class is particularly useful 😄

Reply:
I think I want to have a registry that maps type -> event executor, for event execution